src/pyOpenMS/pxds/ProteaseDigestion.pxd

```diff
@@ -26,12 +26,28 @@ cdef extern from "<OpenMS/CHEMISTRY/ProteaseDigestion.h>" namespace "OpenMS":
         #
         #         dig = ProteaseDigestion()
         #         dig.setEnzyme('Lys-C')
-        #         bsa = "".join([l.strip() for l in open("bsa.fasta").readlines()[1:]])
-        #         bsa = AASequence.fromString(bsa)
-        #         result = []
-        #         dig.digest(bsa, result)
-        #         print(result[4].toString()) # GLVLIAFSQYLQQCPFDEHVK
-        #         print(len(result)) # 57 peptides
+        #         bsa_string = "".join([l.strip() for l in open("bsa.fasta").readlines()[1:]])
+        #         bsa_oms_string = String(bsa_string) # convert python string to OpenMS::String for further processing
+        #
+        #         minlen = 6
+        #         maxlen = 30
+        #
+        #         # Using AASequence and digest
+        #         result_digest = []
+        #         result_digest_min_max = []
+        #         bsa_aaseq = AASequence.fromString(bsa_oms_string)
+        #         dig.digest(bsa_aaseq, result_digest)
+        #         dig.digest(bsa_aaseq, result_digest_min_max, minlen, maxlen)
+        #         print(result_digest[4].toString()) # GLVLIAFSQYLQQCPFDEHVK
+        #         print(len(result_digest)) # 57 peptides
+        #         print(result_digest_min_max[4].toString()) # LVNELTEFAK
+        #         print(len(result_digest_min_max)) # 42 peptides
+        #
+        #         # Using digestUnmodified without the need for AASequence from the EnzymaticDigestion base class
+        #         result_digest_unmodified = []
+        #         dig.digestUnmodified(StringView(bsa_oms_string), result_digest_unmodified, minlen, maxlen)
```

Comment:
hmm the use of StringView in python is a bit ugly.

Reply:
I assume you will need a char* overload for the StringView ctor and a special conversion provider that accepts python bytes (immutable) or bytearrays (mutable) whenever StringView is encountered.